test/system/075-exec.bats

```diff
@@ -166,4 +166,38 @@ load helpers
     run_podman rm -f -t0 $cid
 }
 
+
+@test "podman exec --tty" {
+    run_podman run -d --name test $IMAGE top
+    run_podman 1 exec test printenv echo $TERM
+    is "$output" "" "TERM should not be set"
+    run_podman exec -t test printenv TERM
+    is "$output" "xterm.*" "TERM should be set to xterm"
+    run_podman exec -t --env TERM=foobar test printenv TERM
+    is "$output" "foobar.*" "TERM should be set to xterm"
+    if ! is_remote; then
+        TERM=apple run_podman exec -t test printenv TERM
+        is "$output" "apple.*" "TERM should be set to xterm"
+    fi
+    run_podman rm -f -t 0 test
+
+    run_podman run -t -d --name test $IMAGE top
```

Comment:
This entire block looks a duplicate of the previous one.

Reply:
Way too many confusing combinations. Might I make a suggestion?
```bash
@test "podman exec --tty" {
    # Outer loops: different variations on the RUN container
    for run_opt_t in "" "-t"; do
        for run_term_env in "" "explicit_RUN_term"; do
            local run_opt_env=
            if [[ -n "$run_term_env" ]]; then
                run_opt_env="--env=TERM=$run_term_env"
            fi
            run_podman run -d $run_opt_t $run_opt_env --name test $IMAGE top

            # Inner loops: different variations on EXEC
            for exec_opt_t in "" "-t"; do
                for exec_term_env in "" "explicit_EXEC_term"; do
                    # What to expect.
                    local expected=
                    # "exec -t" defaults to current $TERM
                    if [[ -n "$exec_opt_t" ]]; then
                        expected="$TERM"
                    fi
                    # ...unless "run -t", which forces xterm (always xterm)
                    if [[ -n "$run_opt_t" ]]; then
                        expected="xterm"
                    fi
                    # ...unless overridden by explicit --env
                    if [[ -n "$run_term_env$exec_term_env" ]]; then
                        # (exec overrides run)
                        expected="${exec_term_env:-$run_term_env}"
                    fi

                    local exec_opt_env=
                    if [[ -n "$exec_term_env" ]]; then
                        exec_opt_env="--env=TERM=$exec_term_env"
                    fi

                    run_podman exec $exec_opt_t $exec_opt_env test sh -c 'echo -n $TERM'
                    assert "$output" = "$expected" "run $run_opt_t $run_opt_env, exec $exec_opt_t $exec_opt_env"
                done
            done

            run_podman rm -f -t0 test
        done
    done
```
This very clearly demonstrates what I consider a confusing aspect of `run -t` with `exec -t`. If it's deliberate, fine, but to me it violates POLA that `exec -t` depends on whether the container was `run` with/without `-t`.

Reply:
This is fine, it does loose the test to make sure TERM is not set in the non `run -t` and non `exec -t` case, since theoretically TERM="" versus TERM not set. But really not important.

Reply:
That would be fairly easy to fix:
```bash
    local expect_rc=0
    if [[ -z "$expect_output" ]]; then
         expect_rc=1
    fi
    ...
    run_podman $expect_rc exec [...] printenv TERM
```
...but that then requires defining `cr=$'\r'` and requiring it in the `-t` cases. Which I think might be fine, but it makes the test more complicated, and the loop is already too unwieldy. LMK if you'd like the full `\r` set of changes.